server/aesdsocket.c

```diff
@@ -121,7 +123,28 @@ void write_to_file(int fd, void *buf, size_t buf_size)
     #endif
     ssize_t bytes_written = 0;
     int len = buf_size;
+    char *sock_seek = "AESDCHAR_IOCSEEKTO:";
+    uint32_t write_cmd, write_cmd_offset;
+    struct aesd_seekto seekto;
     
+    #ifdef USE_AESD_CHAR_DEVICE
+    if(strncmp((const char *)buf, (const char *)sock_seek, strlen(sock_seek)) == 0)
+    {
+        sscanf( buf, "AESDCHAR_IOCSEEKTO:%d,%d", &write_cmd, &write_cmd_offset);
```

Comment:
@PeterBraganza you need error handling for sscanf here to handle malformed input.